src/CMakeLists.txt

```diff
@@ -462,6 +466,10 @@ ADD_SUBDIRECTORY ( Commands )
 ADD_SUBDIRECTORY ( Desktop )
 ADD_SUBDIRECTORY ( CommandLine )
 ADD_SUBDIRECTORY ( Tests )
+if (CMAKE_COMPILER_IS_GNUCC)
```

Comment:
The way I see it, cmake doesn't need to know what the architecture actually is, just whether the cpu features and simd implementations support it (though my proposed method assumes those two pieces are developed to support the same set of platforms).  I don't feel that strongly about which method is used, though.

I would also be tempted to instead use those preprocessor defines as guards around the cpuinfo and dot implementations, to make them compile and behave reasonably on all architectures (on unsupported arch, return no features supported, and use a naive dot implementation), but you don't have to do this.